packages/glimmer-compiler/lib/javascript-compiler.ts

```diff
@@ -182,7 +182,13 @@ export default class JavaScriptCompiler {
   /// Expressions
 
   literal(value: any) {
-    this.pushValue(value);
+    if (value === 'undefined') {
+      this.pushValue(JSON.stringify("undefined"));
```

Comment:
Overloading strings wouldn't work – now if you have an incoming string `"undefined"` you would have a conflict. This also unnecessarily complicates decoding strings; I think we should just go with `{ "undefined": true }` or `{ "literal": "undefined" }`. This is rarely used (and if is it would be gzipped just fine), and we have bought a lot of headroom for wire format size, so I'm not too worried about it.